2.5.3 release newspost (#406)
mixxxdj/website: content/news/XXXX-XX-XX-mixxx-2_5_3-released.md

Mixxx has integrated the [xwax](https://xwax.org/overview.html) library for timecode support, which uses an [Alpha-Beta Filter](https://en.wikipedia.org/wiki/Alpha_beta_filter) for pitch detection. This filter provides a good feeling during scratching, but isn't able to provide a stable pitch during normal playback.
In [PR #15194](https://github.com/mixxxdj/mixxx/pull/15194) developers have replaced the Alpha-Beta Filter with a more advanced [Kalman-Filter](https://en.wikipedia.org/wiki/Kalman_filter) equivalent.

Kalman-Filters have many application as GPS navigation and weather forecast models. In Mixxx the Kalman-Filter uses a [probabilistic](https://en.wikipedia.org/wiki/Probability) model of the vinyl movements to predict the velocity, very similar to the Alpha-Beta Filter. This predicted speed is compared with the noisy crackling input signal. The deviation [^2] is fed back into the filter and slightly alters its internal state for the next prediction.

> If the vinyl is, say, spinning at 33 1/3 r/min, that value is used as the initial assumption on which the next prediction is based. If the next measurement returns 40 r/min, the value is not fully trusted, because such a high acceleration is unlikely. This measurement will be used with a low trust. However, if the following measurements detect 40 r/min as well, the filter gradually adjusts to to approximate 40 r/min in further predictions.

Reviewer: ```suggestion
If the vinyl is, say, spinning at 33 1/3 r/min, that value is used as the initial assumption on which the next prediction is based. If the next measurement returns 40 r/min, the value is not fully trusted, because such a high acceleration is unlikely. This measurement will be used with a low trust. However, if the following measurements detect 40 r/min as well, the filter gradually adjusts to approximate 40 r/min in further predictions.
```

Reviewer: Looks good now except this typo!

title: Mixxx 2.5.3 Released
authors: Evelynne Veys
tags: 2.5.3, release announcement
comments: yes
status: draft

#### Mixxx 2.5.3 Release Announcement: Back to School, back to DVS.

> We're proud to announce a new stable release of Mixxx: version 2.5.3, this version not only contains updates and fixes for some issues reported by users, but it also contains a major Digital Vinyl System (DVS) rework. This rework is the outcome of some intensive holiday work and some bright new insights of developers spinning the wheel instead of sunbathing.

Reviewer: ```suggestion
We're proud to announce a new stable release of Mixxx: version 2.5.3, this version not only contains updates and fixes for some issues reported by users, but it also contains improvements to the Digital Vinyl System (DVS) support. These improvements are the result of some bright new insights of developers spinning the wheel instead of sunbathing.
```

> Because of the big improvements we have chosen to backport [^1] it to the 2.5 series instead of integrating it in future alfa & beta versions.

Reviewer: ```suggestion
Since these improvements drastically improve the vinyl experience, we have chosen to backport [^1] it to the 2.5 series instead of integrating it in future alpha & beta versions.
```

Have a nice Mixxx.

[^1]: Back porting is integrating new work into previous releases in order to let those releases profit from the new development.

#### Digital Vinyl System - Timecode System.

> A good explanation about Digital Vinyl Systems and Timecode can be found in [this article on the site of Digital DJ Tips](https://www.digitaldjtips.com/a-beginners-guide-to-digital-vinyl-systems/), and in our own Newsarchive: [How Does Timecode Vinyl Actually Work? (Pt. 1)](https://mixxx.org/news/2021-11-21-dvs-internals-pt1/) and [How Does Timecode Vinyl Actually Work? (Pt. 2)](https://mixxx.org/news/2021-12-22-dvs-internals-pt2/).

Reviewer: ```suggestion
An introduction to Digital Vinyl Systems and timecode can be found in [this article on the site of Digital DJ Tips](https://www.digitaldjtips.com/a-beginners-guide-to-digital-vinyl-systems/), and in our own news archive: [How Does Timecode Vinyl Actually Work? (Pt. 1)](https://mixxx.org/news/2021-11-21-dvs-internals-pt1/) and [How Does Timecode Vinyl Actually Work? (Pt. 2)](https://mixxx.org/news/2021-12-22-dvs-internals-pt2/).
```

> An introduction to Digital Vinyl Systems and timecode can be found in [this article on the site of Digital DJ Tips](https://www.digitaldjtips.com/a-beginners-guide-to-digital-vinyl-systems/), and in our own news archive: [How Does Timecode Vinyl Actually Work? (Pt. 1)](https://mixxx.org/news/2021-11-21-dvs-internals-pt1/) and [How Does Timecode Vinyl Actually Work? (Pt. 2)](https://mixxx.org/news/2021-12-22-dvs-internals-pt2/).

Reviewer 2: use internal links when linking to internal content
```suggestion
An introduction to Digital Vinyl Systems and timecode can be found in [this article on the site of Digital DJ Tips](https://www.digitaldjtips.com/a-beginners-guide-to-digital-vinyl-systems/), and in our own news archive: [How Does Timecode Vinyl Actually Work? (Pt. 1)]({filename}/news/2021-11-21-dvs-internals-pt1.md) and [How Does Timecode Vinyl Actually Work? (Pt. 2)]({filename}/news/2021-12-22-dvs-internals-pt2.md).
```

> To summarize, Mixxx plays the song by following the signal on the timecode media, which gets internally converted to actual timeframes via the LUT.

Reviewer 2: ```suggestion
To summarize, Mixxx plays the song by following the signal on the timecode media, which gets internally converted to actual timeframes via a lookup table (LUT).
```
the term hasn't been introduced at this point in the document yet.

> In short summarized: Mixxx plays the music tracks by following the input signal produced by turntables/CD-players (and other Timecode-mediaplayers) which gets internally converted to actual timeframes by looking up references to timeframes in the musicfile.

Reviewer: ```suggestion
To summarize, Mixxx plays the song by following the signal on the timecode media, which gets internally converted to actual timeframes via the LUT.
```

> So the DJ controls the playback of the digital music by manipulating vinyl or CDs.

Reviewer: ```suggestion
Mixxx plays the song by following the signal on the timecode media, which gets internally converted to actual timeframes. So the DJ controls the playback of the digital music by manipulating vinyl or CDs.
```

> In order to create a vinyl-like feeling the song needs to follow the signal as closely as possible (low latency, fast responsiveness) for e.g. scratching or backspins. There is no 'standard' signal - each manufacturer uses its own code. The demodulated code from the signal needs to be converted to a usable timecode. This is realized by storing the states of the code in a LUT which is mapped to instants in time in the song. This makes it possible to quickly jump to positions in time during playback.

Reviewer 3: We should explain that there are two signals: an audible one to measure velocity (the tone), and the digital one to determine timecode position.  The code must use both of them together to generate smooth and responsive playback.

Reviewer 4: > The code must use both of them together to generate smooth and responsive playback.

I am afraid that this is not the case right now when determining the pitch. But this is a good opportunity for augmenting the Kalman-Filter in a future iteration.

Reviewer 3: The code does use both -- it's always using the pitch to determine speed and direction, and then uses the timecode to determine syncing / seeking. A lot of the trickiness of making vinyl control work is deciding when to use one or the other and how to correct for errors.  I did not say it combines the two "when determining the pitch".  I made this note because the current text makes it sound like the digital timecode is the only signal, which leaves out the analog tone aspect.

Reviewer: Technically the code doesn't play a role in any of those features. We don't really have to explain the code here at all. It has been done in other posts. I should have maybe bothered more with correctness here, but I am quite exhausted atm.

Author: Should we add a new layer of technicality?
In this release post we added  already quite some techinical terms, I'd like to keep it kind of 'low entry'.
I am confident that readers who wish to delve further into the subject will find the full information in the other posts.

Reviewer 3: Then maybe drop the bit about LUTs etc.  Just say "In order to create a vinyl-like feeling the song needs to follow the signal as closely as possible (low latency, fast responsiveness) for e.g. scratching or backspins. There is no 'standard' signal - each manufacturer uses its own code. The demodulated code from the signal needs to be converted to a usable timecode. "

Reviewer: I requested some changes. The new formulations more accurately describe what is happening and why the new PRs make a difference.

> In order to create a real vinyl-handling-feeling the produced music needs to follow the inputsignal as close as possible (low latency, fast responsiveness) for eg scratching, spinbacks etc. As there is no 'standard' inputsignal each manufacturer uses its own code. The received inputsignalcode needs to be converted to a usable timecode by using a LUT (Look Up Table), for each manufacturer another LUT needs to be created and used.  

Reviewer: ```suggestion
In order to create a vinyl-like feeling the song needs to follow the signal as closely as possible (low latency, fast responsiveness) for e.g. scratching or backspins. There is no 'standard' signal - each manufacturer uses its own code. The demodulated code from the signal needs to be converted to a usable timecode. This is realized by storing the states of the code in a LUT (lookup table) which is mapped to instants in time in the song. This makes it possible to quickly jump to positions in time during playback.
```

> In order to create a vinyl-like feeling the song needs to follow the signal as closely as possible (low latency, fast responsiveness) for e.g. scratching or backspins. There is no 'standard' signal - each manufacturer uses its own code. The demodulated code from the signal needs to be converted to a usable timecode. This is realized by storing the states of the code in a LUT (lookup table) which is mapped to instants in time in the song. This makes it possible to quickly jump to positions in time during playback.

Reviewer 2: now that it has been introduced, you can remove it.
```suggestion
In order to create a vinyl-like feeling the song needs to follow the signal as closely as possible (low latency, fast responsiveness) for e.g. scratching or backspins. There is no 'standard' signal - each manufacturer uses its own code. The demodulated code from the signal needs to be converted to a usable timecode. This is realized by storing the states of the code in a LUT which is mapped to instants in time in the song. This makes it possible to quickly jump to positions in time during playback.
```

> The conversion of the inputsignal to the actual timeframecode is handled by the [XWAX library](https://xwax.org/overview.html). To filter 'impossible' or not 'logical' signals (eg frame 7000 directly after frame 1000) the [alpha-beta filter](https://en.wikipedia.org/wiki/Alpha_beta_filter) was used.

Reviewer: ```suggestion
The conversion from timecode signal to actual timeframe is handled by the [xwax](https://xwax.org/overview.html) library. To detect the pitch of the signals (e.g. the playback speed) xwax uses an [Alpha-Beta Filter](https://en.wikipedia.org/wiki/Alpha_beta_filter).
```

> The conversion from timecode signal to actual timeframe is handled by the [xwax](https://xwax.org/overview.html) library. To detect the pitch of the signals (e.g. the playback speed) xwax uses an [Alpha-Beta Filter](https://en.wikipedia.org/wiki/Alpha_beta_filter).

Reviewer: ```suggestion
In order to create a vinyl-like feeling, the song needs to follow the signal as closely as possible (low latency, fast responsiveness) for e.g. scratching or backspins. There is no 'standard' signal - each manufacturer uses its own code. One thing all manufacturers have in common is that the code is modulated onto a sine wave. The code is used for absolute positions e.g. after needle drops, but it is not suitable to handle pitch (e.g. the playback speed) or playback direction. For this the sine wave needs to be processed by a filter that detects phase differences and converts it into pitch information. 
```

> In the rework with [PR #15194](https://github.com/mixxxdj/mixxx/pull/15194) developers have replaced the alpha-beta filter with the [Kalman-filter](https://en.wikipedia.org/wiki/Kalman_filter). Kalman filters are generally used in GPS navigation and weather forecast models.

Reviewer: ```suggestion
In [PR #15194](https://github.com/mixxxdj/mixxx/pull/15194) developers have replaced the Alpha-Beta Filter with a more advanced [Kalman-Filter](https://en.wikipedia.org/wiki/Kalman_filter) equivalent. Kalman-Filters are generally used in GPS navigation and weather forecast models.
```

> In [PR #15194](https://github.com/mixxxdj/mixxx/pull/15194) developers have replaced the Alpha-Beta Filter with a more advanced [Kalman-Filter](https://en.wikipedia.org/wiki/Kalman_filter) equivalent. Kalman-Filters are generally used in GPS navigation and weather forecast models.

Reviewer: ```suggestion
Mixxx has integrated the [xwax](https://xwax.org/overview.html) library for timecode support, which uses an [Alpha-Beta Filter](https://en.wikipedia.org/wiki/Alpha_beta_filter) for pitch detection. This filter provides a good feeling during scratching, but isn't able to provide a stable pitch during normal playback.
In [PR #15194](https://github.com/mixxxdj/mixxx/pull/15194) developers have replaced the Alpha-Beta Filter with a more advanced [Kalman-Filter](https://en.wikipedia.org/wiki/Kalman_filter) equivalent.
```

> The use of it in DVS in short: a Kalman filter maintains a model of the vinyl to predict the current pitch. This predicted pitch is compared with the noisy crackling input signal. The deviation [^2] determines the trust which is used to incorporated these values into the model for the next prediction.

Reviewer: ```suggestion
The Kalman-Filter reuses the current constant velocity model of the Alpha-Beta Filter to estimate the pitch, but offers some improvements in pitch stability. The model compares pitch predictions based on past values with the somewhat noisy measurements. The deviation [^2] determines the trust which is used to incorporated these values into the model for the next prediction.
```

Reviewer 4: I think the we should keep the "model of the vinyl" term. It is probably over-simplified, because we look only at the speed, but its very illustrative. It is a common explanation that a Kalman filter maintains a model of something and instead of using the sensor values directly, they are used for augmentation of such a model. The output value is read from the internal model (of the vinyl in our case) not form the input values. With this explanation I was successful to explain it even to collegues without a technical beackground.  

```suggestion
The use of it in DVS in short: a Kalman-Filter filter maintains a model of the vinyl to predict the current pitch. This predicted pitch is compared with the noisy crackling input signal. The deviation [^2] determines the trust which is used to incorporated these values into the model for the next prediction. It reuses the current constant velocity model of the original Alpha-Beta Filter.
```

> The use of it in DVS in short: a Kalman-Filter maintains a model of the vinyl to predict the current pitch. This predicted pitch is compared with the noisy crackling input signal. The deviation [^2] determines the trust which is used to incorporate these values into the model for the next prediction. It reuses the current constant velocity model of the original Alpha-Beta Filter.

Reviewer: ```suggestion
Kalman-Filters have many application as GPS navigation and weather forecast models. In Mixxx the Kalman-Filter uses a probabalistic model to predict velocity changes, very similar to the Alpha-Beta Filter. This predicted pitch is compared with the noisy crackling input signal. The deviation [^2] is fed back into the filter and slightly alters its internal state for the next prediction. 
```

Reviewer 4: "probabilistisches Modell" :-/ This might be correct, but you loose readers here (see my earlier comment). I don't know what this is. 

https://dictionary.cambridge.org/dictionary/english/probabilistic

How about: 

```suggestion
Kalman-Filters have many application as GPS navigation and weather forecast models. In Mixxx the Kalman-Filter uses a probabalistic model of the vinyl movements to predict the velocity, very similar to the Alpha-Beta Filter. This predicted speed is compared with the noisy crackling input signal. The deviation [^2] is fed back into the filter and slightly alters its internal state for the next prediction. 
```
With or without the term "probabalistic"

Reviewer 2: Just [probabalistic](https://en.wikipedia.org/wiki/Probability) with a link?

> 

Reviewer: ```suggestion
```

Author: used combination with link to https://en.wikipedia.org/wiki/Probability

Reviewer: Nice

> An example: The vinyl is likely rotating with 33 1/3 r/min. This is used as a first prediction. If the next measurement is 40 r/min the value is not trust worthy because such a speed-up is unlikely. This measurement will be used with a low trust. If all following values are however at 40 r/min the filter looses trust for the predicted value and slowly adopts 40 r/min for further predictions.

Reviewer: ```suggestion
If the vinyl is, say, spinning at 33 1/3 r/min, that value is used as the initial assumption on which the next prediction is based. If the next measurement returns 40 r/min, the value is not fully trusted, because such a high acceleration is unlikely. This measurement will be used with a low trust. However, if the following measurements detect 40 r/min as well, the filter gradually adjusts to to approximate 40 r/min in further predictions.
```

> Using this deviation in Mixxx helps to properly and more accurately represent eg the rate-slider on the turntable/CD-player.

Reviewer: ```suggestion
By using this model, Mixxx is able to properly and more accurately represent e.g. the pitch control slider on the turntable or CD-player.
```

> Another improvement has been made to the measurements which are fed to the Kalman-Filter [PR 15217](https://github.com/mixxxdj/mixxx/pull/15217). The current model only takes rough measurements of the position on the sine wave. Furthermore these measurements work on the assumption to always be correct. A new check was added to detect if measurements were skipped or larger than assumed. This improves backspins where the deviation of estimation and measurement suddenly become very high.

Reviewer 2: same syntax as above for consistency
```suggestion
Another improvement has been made to the measurements which are fed to the Kalman filter [PR #15217](https://github.com/mixxxdj/mixxx/pull/15217). The current model only takes rough measurements of the position on the sine wave. Furthermore these measurements work on the assumption to always be correct. A new check was added to detect if measurements were skipped or larger than assumed. This improves backspins where the deviation of estimation and measurement suddenly become very high.
```

These changes are major improvements for all DVS users. Too good to wait for Mixxx 2.6.  

> Get all Mixxx-ed up and join our [testing-force](https://mixxx.org/get-involved/) for more upcoming features.

Reviewer 2: same here
```suggestion
Get all Mixxx-ed up and join our [testing-force]({filename}/pages/get-involved.md) for more upcoming features.
```

> [^2]: The Deviation in Kalman filters is called [Innovation](https://en.wikipedia.org/wiki/Innovation_(signal_processing).

Reviewer 2: Missing closing ). Link doesn't get recognized.

```suggestion
[^2]: The Deviation in Kalman filters is called [Innovation](https://en.wikipedia.org/wiki/Innovation_(signal_processing)).
```

Author: aah .. I'm blind

[^2]: The Deviation in Kalman filters is called [Innovation](https://en.wikipedia.org/wiki/Innovation_(signal_processing).

### Controller Mappings

Mappings were updated for

> * Icon P1-Nano MIDI 1
> * Traktor Kontrol S4 Mk3
> * Traktor Kontrol S3
> * Numark NS6II

Reviewer 2: otherwise the unordered list is not recognized as such and not properly rendered
```suggestion

* Icon P1-Nano MIDI 1
* Traktor Kontrol S4 Mk3
* Traktor Kontrol S3
* Numark NS6II
```

title: Mixxx 2.5.3 Released
authors: Evelynne Veys
tags: 2.5.3, release announcement
comments: yes
status: draft

#### Mixxx 2.5.3 Release Announcement: Back to School, back to DVS.

We're proud to announce a new stable release of Mixxx: version 2.5.3, this version not only contains updates and fixes for some issues reported by users, but it also contains a major Digital Vinyl System (DVS) rework. This rework is the outcome of some intensive holiday work and some bright new insights of developers spinning the wheel instead of sunbathing.
Because of the big improvements we have chosen to backport [^1] it to the 2.5 series instead of integrating it in future alfa & beta versions.

Have a nice Mixxx.

[^1]: Back porting is integrating new work into previous releases in order to let those releases profit from the new development.

#### Digital Vinyl System - Timecode System.

A good explanation about Digital Vinyl Systems and Timecode can be found in [this article on the site of Digital DJ Tips](https://www.digitaldjtips.com/a-beginners-guide-to-digital-vinyl-systems/).

> In short summarized: Mixxx plays the music tracks by following the input signal produced by turntables/CD-players (and other Timecode-mediaplayers), so the DJ controls the playback of the digital music by manipulating vinyl or CDs.

Reviewer 4: We should also link our own articles https://mixxx.org/news/2021-11-21-dvs-internals-pt1/ https://mixxx.org/news/2021-11-21-dvs-internals-pt2/

In order to create a real vinyl-handling-feeling the produced music needs to follow the inputsignal as close as possible (low latency, fast responsiveness) for eg scratching, spinbacks etc.

> In the past Mixxx tried to follow the input signal as accurately as possible, as there came some updates in Timecodemedia (producing a lot more info), this logic needed to be updated.

Reviewer 2: not true (see previous point), we already used xwax's (compared to the kalman filter) suboptimal alpha-beta filter).

Author: I agree but... before we got lost in translation explaining the details of the changes.
I wanted to give an idea to readers without drowning them in a tidal wave of technical & mathematical equations.
(and explaining the xwax)
Can you suggest a better description?

(you were fast 😄 )

Reviewer 2: The description of the kalman filter is fine, just how its used is incorrect. We don't make more predictions with the kalman filter, the predictions are just less-noisy than those of the alpha-beta filter.

There is also no need to explain xwax entirely, just say that its the library we use to do DVS and that we made those two changes to it (adding the proper phase tracker and adding an improved filter).

Author: I'll re-orden things in my mind and will push a re-write tonight.

By the way: 
1. do you now why the link in the footnote is not working or are links not possible in footnotes?
2. I tried to strikethrough school like ~~school.~~  but that didn't work :-(

and last but not least: thank you for your review/co-operation.

Developers have seen the light by choosing the introduction of a [Kalman-filter](https://en.wikipedia.org/wiki/Kalman_filter).
In short: a Kalman filter can be used to determine the deviation of the actual (observed) point in regard to the pre-calculated point.
An example: if a train starts at point A at 12:00 to end at point B, 120km further, at a speed of 120 km/h you can calculate all intermediate points and the estimated time of arrival (ETA):
30km at 12:15, 60km at 12:30, 90km at 12:45 ... Point B at 13:00.

> If the driver uses a GPS, it will start with ETA of 13:00. When the GPS notices that the position of the train at 12:15 is different from the pre-calculated position, eg 28km or 32km,  the GPS knows the train has a higher/lower speed and can recalculate the intermediate points/ETA and display another speed.

Reviewer 4: Let me try to improve it. 

```suggestion
A Kalman filter maintains a model of the vinyl to predict the current pitch. This predicted pitch is compared with the noisy crackling input signal. The deviation determines the trust which is used to incorporated these values into the model for the next prediction.

An example: The vinyl is likely rotating with 33 1/3 r/min. This is used as a first prediction. If the next measurement is 40 r/min the value is not trust worthy because such a speed-up is unlikely. This measurement will be used with a low trust. If all following values are however at 40 r/min the filter looses trust for the predicted value and slowly adopts 40 r/min for further predictions. 
```

Author: Thank you, that's a better example, that can be qualified as low-entry too. TOP!

Another example: Kalman filters are also used in weather forecasts to predict the time a storm passes at a certain point (place).

Instead of calculating a couple of points on the trajectories thousands of points can be calculated (and even more), the more pre-calculated and compared points the more accurate the calculation of the deviation [^2] will be.

With the Kalman filter Mixxx can calculate the deviation between the observed point (input from Timecodesignal)  and the pre-calculated point.
Using this deviation in Mixxx helps Mixxx to properly and more accurately represent eg the rate-slider on the turntable/CD-player.

This is a major improvement useful for everybody, not only for the alpha/beta-users.

> Get all Mixxx-ed up and join our [testing-force](https://mixxx.org/get-involved/).  

Reviewer 2: I don't think this summarizes the changes well. 
* #15194 introduced the kalman filter as the new default. There already previously was still a filter ("alpha-beta filter"). The kalman filter is better at producing a stable output while maintaining responsiveness. 
* #15217 Tries to improve sticker drift by trying to account for needle skips.

Reviewer 4: For #15217 

In addition we have added a plausibility check for the time code signal. This compares the left and right channel of the time code signal to remove noise and is even able to restore the original information. This significantly reduces sticker drift during fast scratch actions.

So the DJ controls the playback of the digital music by manipulating vinyl or CDs.

In order to create a real vinyl-handling-feeling the produced music needs to follow the inputsignal as close as possible (low latency, fast responsiveness) for eg scratching, spinbacks etc. As there is no 'standard' inputsignal each manufacturer uses its own code. The received inputsignalcode needs to be converted to a usable timecode by using a LUT (Look Up Table), for each manufacturer another LUT needs to be created and used.  
The conversion of the inputsignal to the actual timeframecode is handled by the [XWAX library](https://xwax.org/overview.html). To filter 'impossible' or not 'logical' signals (eg frame 7000 directly after frame 1000) the [alpha-beta filter](https://en.wikipedia.org/wiki/Alpha_beta_filter) was used.

In the rework with [PR #15194](https://github.com/mixxxdj/mixxx/pull/15194) developers have replaced the alpha-beta filter with the [Kalman-filter](https://en.wikipedia.org/wiki/Kalman_filter). Kalman filters are generally used in GPS navigation and weather forecast models.

> The use of it in DVS in short: a Kalman filter maintains a model of the vinyl to predict the current pitch. This predicted pitch is compared with the noisy  cracklinginput signal. The deviation [^2] determines the trust which is used to incorporated these values into the model for the next prediction.

Reviewer 4: Whitspace issue: 
```suggestion
The use of it in DVS in short: a Kalman filter maintains a model of the vinyl to predict the current pitch. This predicted pitch is compared with the noisy crackling input signal. The deviation [^2] determines the trust which is used to incorporated these values into the model for the next prediction.
```

An example: The vinyl is likely rotating with 33 1/3 r/min. This is used as a first prediction. If the next measurement is 40 r/min the value is not trust worthy because such a speed-up is unlikely. This measurement will be used with a low trust. If all following values are however at 40 r/min the filter looses trust for the predicted value and slowly adopts 40 r/min for further predictions.

Using this deviation in Mixxx helps to properly and more accurately represent eg the rate-slider on the turntable/CD-player.

Another improvement has been made with the addition of a plausibility check for the time code signal [PR 15217](https://github.com/mixxxdj/mixxx/pull/15217).

> This compares the left and right channel of the time code signal to remove noise and is even able to restore the original information. This significantly reduces sticker drift during fast scratch actions. (see [How Does Timecode Vinyl Actually Work? (Pt. 2)](https://mixxx.org/news/2021-12-22-dvs-internals-pt2/) #2)

Reviewer: ```suggestion
Another improvement has been made to the measurements which are fed to the Kalman filter [PR 15217](https://github.com/mixxxdj/mixxx/pull/15217). The current model only takes rough measurements of the position on the sine wave. Furthermore these measurements work on the assumption to always be correct. A new check was added to detect if measurements were skipped or larger than assumed. This improves backspins where the deviation of estimation and measurement suddenly become very high.
```

Author: So we don't mention the sticker drift? We only mention backspins?

> These are major improvements useful for everybody, not only for the alpha/beta-users.

Reviewer 4: I struggle a bit with this sentences, because the backport topic is mentioned already at the top of the paragraph.  

```suggestion
These changes are major improvements for all DVS users. Too good to wait for Mixxx 2.6.  
```

> Get all Mixxx-ed up and join our [testing-force](https://mixxx.org/get-involved/).  

Reviewer 4: ```suggestion
Get all Mixxx-ed up and join our [testing-force](https://mixxx.org/get-involved/) for more upcoming features.
```

[^2]: The Deviation in Kalman filters is called [Innovation](https://en.wikipedia.org/wiki/Innovation_(signal_processing).

#### Other important updates and fixes in 2.5.3

- Blablabla
- Blablabla
- Blablabla

#### 2.5.3 Changelog
The complete changelog can be found [here](https://github.com/mixxxdj/mixxx/blob/2.5.3/CHANGELOG.md)

### Controller Mappings

Mappings were updated for ... Blablabla

### Blablabla

Blablabla

### Misc

> - Blablabla

Reviewer 4: How about condense the rest to a minimum: 

```suggestion
### Controller Mappings

Mappings were updated for
* Icon P1-Nano MIDI 1
* Traktor Kontrol S4 Mk3
* Traktor Kontrol S3
* Numark NS6II

#### 2.5.3 Changelog

More fixes have been included to keep everything stable, up and running. 
The complete changelog can be found [here](https://github.com/mixxxdj/mixxx/blob/2.5.3/CHANGELOG.md)
